src/futhark-test.hs

```diff
@@ -337,20 +336,18 @@ runTests config paths = do
 
   let report = if isTTY then reportInteractive else reportText
       clear  = if isTTY then clearLine else putStr "\n"
-      getResults remaining failed passed =
-        case S.toList remaining of
-          []      -> clear >> return (failed, passed)
-          first:_ -> do
-            report (testCaseProgram first) failed passed $ S.size remaining
-            (test, res) <- takeMVar resmvar
-            let next = getResults $ test `S.delete` remaining
-            case res of
-              Success -> next failed (passed+1)
-              Failure s -> do clear
-                              T.putStrLn (T.pack (testCaseProgram test) <> ":\n" <> s)
-                              next (failed+1) passed
-
-  (failed, passed) <- getResults (S.fromList included) 0 0
+
+      getResults 0         failed passed = clear >> return (failed, passed)
+      getResults remaining failed passed = do
+        (test, res) <- takeMVar resmvar
+        report (testCaseProgram test) failed passed remaining
```

Comment:
I'm not sure this is correct now.  The `report` functions prints text indicating which test we are *waiting* for, but you are now printing the name of the test that was most recently *completed*.  The original motivation for this piece of information was to report tests that were hanging (because of infinite loops or whatnot).

Reply:
I see. It's my misunderstanding then. I thought it was a bug because it was stuck on one test most of the time when I ran the test and I thought it was supposed to print the test it completes. I think it still prints "Waiting for BabyBearFun.fut" (in my case) for a long time though, even if that test may not have started yet.

Reply:
Yes, it's a little confusing.  It would be better if it printed the next test to be run, instead of some arbitrary test.  But I'm not sure that this information is easily available.  You could add another kind of message that is put in the MVar whenever a test is chosen for testing, and then use that to print the message.

Reply:
I think it'll print whatever test is first in the list regardless of whether it has been started by a thread or not.

Reply:
Indeed, there is definitely room for improvement.  But I think it's important to maintain the property that if some test is hanging (or just significantly slower than everything else), it will eventually hang while having its name displayed here.  That's an important feature.

Reply:
You are right. Please just ignore this pull request then.